pkg/agent/watch/watch.go

```diff
@@ -0,0 +1,214 @@
+// Copyright The NRI Plugins Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package watch
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/watch"
+
+	logger "github.com/containers/nri-plugins/pkg/log"
+)
+
+type (
+	EventType = watch.EventType
+	Event     = watch.Event
+	Interface = watch.Interface
+)
+
+const (
+	Added    = watch.Added
+	Modified = watch.Modified
+	Deleted  = watch.Deleted
+	Bookmark = watch.Bookmark
+	Error    = watch.Error
+
+	reopenDelay = 5 * time.Second
+)
+
+// CreateFn creates a watch for the named object.
+type CreateFn func(ctx context.Context, ns, name string) (Interface, error)
+
+// Watch is a wrapper for apimachinery watches that monitor a single object.
+// The wrapper transparently reopens the watch when it expires.
+type Watch struct {
```

Comment:
This is probably balancing between matter of taste and worth the trouble, feel free to disagree and close this comment... But I was wondering if it would make sense to keep the types and consts in this file and split the [object]watch-related types to `object.go` file, that might then be quite similar to `file.go` in structure. Then there would be clearly an object watch, a file watch, and common watch source files.

Reply:
I think it is worth at least a try. We ended up with the current split to `{watch,file}.go` as hiding local file watching behind the same interface was one of the last things I implemented, and the current split also bothered me. I can take a look at it.

Reply:
Also, I think that it might make sense to turn `Agent` into an `interface`...

Reply:
@askervin Updated the watch code splitting the implementation into `{watch,object,file}.go`. I also adjusted the configuration interface slightly (provide an `Unmarshal()` instead of a `WatchFile()`).